projects/python-client/src/datapane/client/api/ipython_utils.py

```diff
@@ -19,10 +25,219 @@ def block_to_iframe(block: BaseElement) -> str:
     Returns:
         HTML string for the block
     """
-    # importing within function scope to avoid circular dependency
     from .report.core import App
 
     app = App(block)
     block_html_string = app.stringify(template_name="ipython_template.html")
 
     return block_html_string
+
+
+@capture_event("Get Jupyter Notebook JSON")
+def get_jupyter_notebook_json() -> dict:
+    """Get the JSON for the current Jupyter notebook
+
+    Returns:
+        Notebook JSON
+    """
+    import ipynbname
+
+    try:
+        nb_path = ipynbname.path()
+        notebook_json = json.loads(open(nb_path, encoding="utf-8").read())
```

Comment:
I think this will keep the file open, generally easier to wrap in `pathlib.Path` and call `path.read_text()` if you want the string, or open the file in a `contextmanager` and use `json.load`

Reply:
Went with context manager